work11/index.js

```diff
@@ -0,0 +1,71 @@
+"use strict";
+const div = document.querySelector(".wrap")
+const ul = document.getElementById("js-list");
+const url = "https://jsondata.okiba.me/v1/json/d4ex5210822124352"
+
+function getData() {
+  const result = new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve(
+        async function () {
+          const response = (await fetch(url)).json();
+          return response
+        }()
+      )
+
+    }, 3000);
+  })
+  return result;
+}
```

Comment:
In the past, we used `setTimeout` pseudo-optimally to get the API, but from now on it is not necessary.

Reply:
This function, but it was not immediately clear what we wanted to do with it. An ideal function is one that you can easily figure out what it wants to do by looking at its contents, so let's try to sort out what it wants to do.

What we want to do with this function
1. Display the loaded image
2. Fetch the data
3. Once the data has been retrieved, convert it to json.
4. Return it when the conversion to json is finished.
5. If an error occurs during the acquisition, display the error on the screen.
6. Turn off loading when processing is complete.

Therefore, it would be better to write this function as `async/await` and `try-catch-finally`.

Reply:
Thank you very much.

You don't have to use setTimeOut to reproduce async because you're getting it using the API, right?

commit number
a5ef11d

Reply:
@443n0511 

What we want to do with this function

1.Display the loaded image
2.Fetch the data
3.Once the data has been retrieved, convert it to json.
4.Return it when the conversion to json is finished.
5.If an error occurs during the acquisition, display the error on the screen.
6.Turn off loading when processing is complete.


Code refactored with reference to the above.
Please check

commit : 524a1d2